src/pages/challenges.tsx

```diff
@@ -1,17 +1,77 @@
-import { FC } from 'react';
+import { FC, useState, useEffect } from 'react';
 import Head from '@/components/head';
 import Layout from '@/components/Layout';
-import Section from '@/components/section';
-import { CHALLENGES_SCREEN } from '@/components/constants/display-sections.js';
-import ChallengesJson from '@/mocks/challenges.json';
-
-const Challenges: FC = () => (
-  <Layout>
-    <Head title="Challenges" />
-    <div>
-      <Section content={ChallengesJson} screen={CHALLENGES_SCREEN} />
-    </div>
-  </Layout>
-);
+import Active from '@/components/challenges/active';
+import Complete from '@/components/challenges/complete';
+import Accordion from '@/components/Accordion';
+import useFetch from '@/hooks/useFetch';
+import { challenge } from '@/components/constants/types';
+import classNames from '@/styles/tasks.module.scss';
 
+const CHALLENGES_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/challenges`;
+
+function renderCardList(challengeSection: challenge['content'], key:string) {
+  if (key === 'Active') {
+    return challengeSection.map((item) => <Active content={item} key={item.id} />);
+  }
+  return challengeSection.map((item) => <Complete content={item} key={item.id} />);
+}
+
+const Challenges: FC = () => {
+  let challenges: challenge['content'] = [];
+
+  const [filteredChallenge, setFilteredChallenge] = useState<any>([]);
+
+  const { response, error, isLoading } = useFetch(CHALLENGES_URL);
+
+  useEffect(() => {
+    if ('challenges' in response) {
+      challenges = response.challenges;
+      const challengeMap: any = [];
+      challenges.forEach((item) => {
+        if (item.is_active) {
+          if ('Active' in challengeMap) {
+            challengeMap.Active = [...challengeMap.Active, item];
+          } else {
+            challengeMap.Active = [item];
+          }
+        } else if (!item.is_active) {
```

Comment:
No need of `else if` here, if the condition in line no 32 fails, it should directly come here
so,
```javascript
else {...your logic}
```